framework/finalizer.go

```diff
@@ -0,0 +1,99 @@
+package framework
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	finalizerName = "operatorkit.giantswarm.io"
```

Comment:
I would like to have this public with proper godoc since this is quite an important piece of public documentation and API reference of how this thing works. Users will observe things like this when working with operatorkit eventually.

Reply:
I don't understand that. This is totally internal thing. Having that in godoc exposed for an end user is arguably confusing.

Reply:
People will see some magic finalizer being added to their CRs and we should have as much as useful docs for that as possible. We do this internally but it affects APIs users are working with. The finalizer is visible to them and not hidden within the go code or something. This is exposed functionality and should be documented.

Reply:
Documented - yes. We can document it at package level, in some external doc files, etc. There are ways to document that. But exporting it has semantic implications and that's why I disagree.

Reply:
Disagree and commit.